pandas/core/arrays/masked.py

```diff
@@ -621,6 +621,18 @@ def _arith_method(self, other, op):
         op_name = op.__name__
         omask = None
 
+        if not hasattr(other, "dtype") and is_list_like(other):
+            # Try inferring masked dtype instead of casting to object
+            inferred_dtype = lib.infer_dtype(other, skipna=True)
+            if inferred_dtype == "integer":
```

Comment:
How about boolean?

Reply:
any particular reason not to call pd.array here?

context: im looking at issues relating to infer_dtype and trying to track down how many of our usages we really need